src/lib/features/project/project-store.ts

```diff
@@ -579,32 +581,84 @@ class ProjectStore implements IProjectStore {
     }
 
     async getApplicationsByProject(
-        projectId: string,
-    ): Promise<IProjectApplication[]> {
+        params: IProjectApplicationsSearchParams,
+    ): Promise<IProjectApplications> {
+        const { project, limit, sortOrder, sortBy, searchParams, offset } =
+            params;
+        const validatedSortOrder =
+            sortOrder === 'asc' || sortOrder === 'desc' ? sortOrder : 'asc';
         const query = this.db
             .with('applications', (qb) => {
                 qb.select('project', 'app_name', 'environment')
                     .distinct()
                     .from('client_metrics_env as cme')
                     .leftJoin('features as f', 'cme.feature_name', 'f.name')
-                    .where('project', projectId);
+                    .where('project', project);
             })
-            .select(
-                'a.app_name',
-                'a.environment',
-                'ci.instance_id',
-                'ci.sdk_version',
-            )
-            .from('applications as a')
-            .leftJoin('client_instances as ci', function () {
-                this.on('ci.app_name', 'a.app_name').andOn(
-                    'ci.environment',
+            .with('ranked', (qb) => {
```

Comment:
This query is intimidating, are there good patterns we can use to make it more succinct and understandable?

Reply:
Refactored part of it into shared util. applySearchFilters